src/Components/Server/src/Circuits/CircuitPersistenceManager.cs

```diff
@@ -21,26 +21,34 @@ internal partial class CircuitPersistenceManager(
 {
     public async Task PauseCircuitAsync(CircuitHost circuit, bool saveStateToClient = false, CancellationToken cancellation = default)
     {
-        var renderer = circuit.Renderer;
-        var persistenceManager = circuit.Services.GetRequiredService<ComponentStatePersistenceManager>();
-        var collector = new CircuitPersistenceManagerCollector(circuitOptions, serverComponentSerializer, circuit.Renderer);
-        using var subscription = persistenceManager.State.RegisterOnPersisting(
-            collector.PersistRootComponents,
-            RenderMode.InteractiveServer);
+        await circuit.Renderer.Dispatcher.InvokeAsync(async () =>
+        {
+            if (circuit.IsDisposed())
+            {
+                return;
+            }
```

Comment:
We shouldn't need to guard against double disposal. If we have this issue, there's a bug somewhere else